module-system/sov-state/src/map.rs

```diff
@@ -115,3 +115,32 @@ where
         working_set.delete_value(self.prefix(), &self.codec, key);
     }
 }
+
+#[cfg(feature = "arbitrary")]
+impl<'a, K, V> StateMap<K, V>
+where
+    K: arbitrary::Arbitrary<'a> + borsh::BorshSerialize + borsh::BorshDeserialize,
+    V: arbitrary::Arbitrary<'a> + borsh::BorshSerialize + borsh::BorshDeserialize,
+{
```

Comment:
You'll have to update this to use `Hash + Eq` for keys when merging `nightly`.

```suggestion
impl<'a, K, V> StateMap<K, V>
where
    K: arbitrary::Arbitrary<'a> + Hash + Eq,
    V: arbitrary::Arbitrary<'a>,
    BorshCodec: StateValueCodec<V>
```

(Having this generic over all codecs and not just `BorshCodec` is likely not worth it and would just complicate things)

I haven't tested this change locally so I'm not sure these type bounds are 100% correct, but should be close.

Reply:
Indeed the change was required with a slight difference from the suggested snippet. Updated!